src/java.base/share/classes/java/lang/invoke/ProxyClassesDumper.java

```diff
@@ -96,10 +97,10 @@ private static void validateDumpDir(Path path) {
         }
     }
 
-    public static String encodeForFilename(String className) {
+    public String encodeForFilename(String className) {
         final int len = className.length();
-        StringBuilder sb = new StringBuilder(len);
-
+        // add some padding to `len` for the index
+        StringBuilder sb = new StringBuilder(len + 5);
```

Comment:
```suggestion
        StringBuilder sb = new StringBuilder(len + 6);
```

Reply:
The index is unlikely to be more than 5 digits (pulled out of a hat really) for most applications I would think; is there something I overlooked here?